pkg/templates/k8sresource_funcs.go

```diff
@@ -60,6 +61,80 @@ func (t *TemplateResolver) fromSecretProtected(namespace string, secretName stri
 	return t.protect(value)
 }
 
+// copies all data in the given Secret, namespace.
+func (t *TemplateResolver) copySecretDataJSON(namespace string, secretname string) (map[string]string, error) {
+	klog.V(2).Infof("copySecretDataJson for namespace: %v, secretname: %v", namespace, secretname)
+
+	ns, err := t.getNamespace("copySecretData", namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	secretsClient := (*t.kubeClient).CoreV1().Secrets(ns)
+	secret, getErr := secretsClient.Get(context.TODO(), secretname, metav1.GetOptions{})
+
+	if getErr != nil {
+		if k8serrors.IsNotFound(getErr) {
+			// Add to the referenced objects if the object isn't found since the consumer may want to watch the object
+			// and resolve the templates again once it is present.
+			t.addToReferencedObjects("/v1", "Secret", ns, secretname)
+		}
+
+		klog.Errorf("Error Getting secret:  %v", getErr)
+		err := fmt.Errorf("failed to get the secret %s from %s: %w", secretname, ns, getErr)
+
+		return nil, err
+	}
+
+	data := make(map[string]string)
+
+	for key, val := range secret.Data {
+		// when using corev1 secret api, the data is returned decoded,
+		// re-encododing to be able to use it in the referencing secret
+		sEnc := base64.StdEncoding.EncodeToString(val)
+		data[key] = sEnc
+	}
+
+	// add this Secret to list of  Objs referenced by the template
+	t.addToReferencedObjects("/v1", "Secret", secret.Namespace, secret.Name)
+
+	return data, nil
+}
+
+// copies all data in the given Secret, namespace.
+func (t *TemplateResolver) copySecretData(namespace string, secretname string) (string, error) {
+	klog.V(2).Infof("copySecretData for namespace: %v, secretname: %v", namespace, secretname)
+
+	data, err := t.copySecretDataJSON(namespace, secretname)
+	if err != nil {
+		return "", err
+	}
+
+	yamlBytes, err := yaml.Marshal(data)
+	if err != nil {
+		return "", fmt.Errorf("failed to convert the policy template to YAML: %w", err)
+	}
+
+	return string(yamlBytes), nil
+}
+
+// copySecretDataProtected wraps copySecretData and encrypts the output value using the "protect" method.
+func (t *TemplateResolver) copySecretDataProtected(namespace string, secretName string) (map[string]string, error) {
```

Comment:
Is there a test for this?

It also seems like all the other methods return a YAML string except for this one.